sympy/core/expr.py

```diff
@@ -3437,15 +3437,13 @@ def compute_leading_term(self, x, logx=None):
         if self.removeO() == 0:
             return self
 
-        from sympy.series.gruntz import calculate_series
-
         if logx is None:
             from .symbol import Dummy
             from sympy.functions.elementary.exponential import log
             d = Dummy('logx')
-            s = calculate_series(expr, x, d).subs(d, log(x))
+            s = expr.series(x, n=1, logx=d).subs(d, log(x))
```

Comment:
n=1 may not suffice for the leading term in general.

Reply:
I've made some changes , which would work as per what we want I guess . Anyways the `compute_leading_term` function was supposed to be deprecated as per #21843 , hence I've added deprecation warnings for the same . I've also removed it's usage completely from the code base .